src/config.ts

```diff
@@ -88,18 +89,29 @@ export async function readDefaultConfig(root?: string): Promise<Config> {
   return normalizeConfig(await importConfig(tsPath), root);
 }
 
+let cachedPages: {key: string; pages: Page[]} | null = null;
```

Comment:
I’m only store a single value (the most recent value) in the cache for `readPages`, just to keep things simple. I don’t expect that there would be more than one set of pages needed simultaneously.